src/components/action-popover/action-popover.test.tsx

```diff
@@ -0,0 +1,2303 @@
+  <ThemeProvider theme={sageTheme}>
+    <ActionPopover {...props}>
+      <ActionPopoverItem href="#" download>
+        test download
+      </ActionPopoverItem>
+    </ActionPopover>
+  </ThemeProvider>
+));
+
+describe("if download prop and href prop are provided", () => {
+  it("should render as a link component", async () => {
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+    render(
+      <ActionPopover>
+        <ActionPopoverItem href="#" download>
+          test download
+        </ActionPopoverItem>
+      </ActionPopover>
+    );
+
+    await user.click(screen.getByRole("button"));
+
+    expect(screen.getByRole("link")).toHaveTextContent("test download");
+  });
+
+  it("should close the menu if enter pressed", async () => {
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+    render(
+      <ActionPopover>
+        <ActionPopoverItem href="#" download>
+          test download
+        </ActionPopoverItem>
+      </ActionPopover>
+    );
+
+    await user.click(screen.getByRole("button"));
+
+    await screen.getByRole("link").focus();
+    await user.keyboard("{Enter}");
+
+    expect(screen.queryByRole("list")).not.toBeInTheDocument();
+  });
+});
+
+test("displays the vertical ellipsis icon as the menu button", () => {
+  render(
+    <ActionPopover>
+      <ActionPopoverItem>example item</ActionPopoverItem>
+    </ActionPopover>
+  );
+  expect(screen.getByTestId("icon")).toHaveStyleRule(
+    "content",
+    `"${iconUnicodes.ellipsis_vertical}"`,
+    { modifier: "&::before" }
+  );
+});
+
+test("has expected border radius", () => {
+  render(
+    <ActionPopover>
+      <ActionPopoverItem>example item</ActionPopoverItem>
+    </ActionPopover>
+  );
+  expect(screen.getByRole("button")).toHaveStyle({
+    borderRadius: "var(--borderRadius050)",
+  });
+});
+
+test("has proper data attributes applied to elements", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+  render(
+    <ActionPopover>
+      <ActionPopoverItem>example item 1</ActionPopoverItem>
+      <ActionPopoverDivider />
+      <ActionPopoverItem>example item 2</ActionPopoverItem>
+    </ActionPopover>
+  );
+
+  await user.click(screen.getByRole("button"));
+
+  expect(screen.getByTestId("action-popover-wrapper")).toHaveAttribute(
+    "data-component",
+    "action-popover-wrapper"
+  );
+  expect(screen.getByRole("list")).toHaveAttribute(
+    "data-component",
+    "action-popover"
+  );
+  const divider = screen.getAllByRole("listitem")[1];
+  expect(divider).toHaveAttribute("data-element", "action-popover-divider");
+});
+
+test("has a default aria-label", () => {
+  render(
+    <ActionPopover>
+      <ActionPopoverItem>example item</ActionPopoverItem>
+    </ActionPopover>
+  );
+  expect(screen.getByRole("button")).toHaveAttribute("aria-label", "actions");
+});
+
+test("uses the aria-label prop if provided", () => {
```

Comment:
I take the point, and am happy to change this - just thought I'd make the point that since we're explicitly testing a prop called `aria-label`, there's an argument for making sure it actually becomes the `aria-label` of the input, rather than something else going on (eg with `aria-labeledby`) to ensure the input has the correct accessible name. In one sense that's just an implementation detail, but as I said if we're providing a prop called `aria-label` I think it would be confusing if it didn't end up as the actual HTML `aria-label` attribute.

That's mostly be just playing devil's advocate though, so having made the point I'm happy to make the change you've suggested 👍